huaweicloud/services/acceptance/gaussdb/data_source_huaweicloud_gaussdb_nosql_flavors_test.go

```diff
@@ -0,0 +1,128 @@
+)
+
+func TestAccGaussDBNoSQLFlavors_basic(t *testing.T) {
+	dataSourceName := "data.huaweicloud_gaussdb_nosql_flavors.test"
+
+	dc := acceptance.InitDataSourceCheck(dataSourceName)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { acceptance.TestAccPreCheck(t) },
+		Providers:    acceptance.TestAccProviders,
+		CheckDestroy: dc.CheckResourceDestroy(),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccGaussDBNoSQLFlavors_basic(),
+				Check: resource.ComposeTestCheckFunc(
+					dc.CheckResourceExists(),
+					resource.TestCheckResourceAttr(dataSourceName, "engine_type", "cassandra"),
+					resource.TestMatchResourceAttr(dataSourceName, "flavors.#", regexp.MustCompile("[1-9]\\d*")),
+				),
+			},
+		},
+	})
+}
+
+func TestAccGaussDBNoSQLFlavors_mongodb(t *testing.T) {
+	dataSourceName := "data.huaweicloud_gaussdb_nosql_flavors.test"
+
+	dc := acceptance.InitDataSourceCheck(dataSourceName)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { acceptance.TestAccPreCheck(t) },
+		Providers:    acceptance.TestAccProviders,
+		CheckDestroy: dc.CheckResourceDestroy(),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccGaussDBNoSQLFlavors_mongodb(),
+				Check: resource.ComposeTestCheckFunc(
+					dc.CheckResourceExists(),
+					resource.TestCheckResourceAttr(dataSourceName, "engine_type", "mongodb"),
+					resource.TestMatchResourceAttr(dataSourceName, "flavors.#", regexp.MustCompile("[1-9]\\d*")),
+				),
+			},
+		},
+	})
+}
+
+func TestAccGaussDBNoSQLFlavors_influxdb(t *testing.T) {
+	dataSourceName := "data.huaweicloud_gaussdb_nosql_flavors.test"
+
+	dc := acceptance.InitDataSourceCheck(dataSourceName)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { acceptance.TestAccPreCheck(t) },
+		Providers:    acceptance.TestAccProviders,
+		CheckDestroy: dc.CheckResourceDestroy(),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccGaussDBNoSQLFlavors_influxdb(),
+				Check: resource.ComposeTestCheckFunc(
+					dc.CheckResourceExists(),
+					resource.TestCheckResourceAttr(dataSourceName, "engine_type", "influxdb"),
+					resource.TestMatchResourceAttr(dataSourceName, "flavors.#", regexp.MustCompile("[1-9]\\d*")),
+				),
+			},
+		},
+	})
+}
+
+func TestAccGaussDBNoSQLFlavors_redis(t *testing.T) {
+	dataSourceName := "data.huaweicloud_gaussdb_nosql_flavors.test"
+
+	dc := acceptance.InitDataSourceCheck(dataSourceName)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { acceptance.TestAccPreCheck(t) },
+		Providers:    acceptance.TestAccProviders,
+		CheckDestroy: dc.CheckResourceDestroy(),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccGaussDBNoSQLFlavors_redis(),
+				Check: resource.ComposeTestCheckFunc(
+					dc.CheckResourceExists(),
+					resource.TestCheckResourceAttr(dataSourceName, "engine_type", "redis"),
+					resource.TestMatchResourceAttr(dataSourceName, "flavors.#", regexp.MustCompile("[1-9]\\d*")),
```

Comment:
`\d` is equivalent to `[0-9]`, use `\\d+` or use `resource.TestCheckResourceAttrSet` instead

Reply:
The regex explain `[1-9]\\d*` means more than one, does not always mathced, but `resource.TestCheckResourceAttrSet(dataSourceName, "flavors.#")` looks better.